src/vstest.console/Internal/ConsoleLogger.cs

```diff
@@ -619,7 +627,39 @@ private void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
         {
             // Stop the progress indicator as we are about to print the summary
             this.progressIndicator?.Stop();
+            SourceSummary summary;
+            var testsPassed = 0;
+            var testsFailed = 0;
+            var testsSkipped = 0;
+            var testsTotal = 0;
             Output.WriteLine(string.Empty, OutputLevel.Information);
+            if (verbosityLevel == Verbosity.Quiet)
+            {
+                foreach (var sd in this.sourceSummaryDictionary.ToArray())
+                {
+                    summary = this.sourceSummaryDictionary[sd.Key];
+
+                    // Failed! Pass {1} Failed {2} skipped {3} Time : 233 se ({4})
+                    if (summary.FailedTests > 0)
+                    {
+                        Output.Information(false, ConsoleColor.Red, string.Format(CultureInfo.CurrentCulture, CommandLineResources.TestRunSummaryFailed, summary.TotalTests, summary.PassedTests, summary.FailedTests, summary.SkippedTests, GetFormattedDurationString(summary.TimeSpan), sd.Key.Split('\\').Last()));
+                    }
+                    else
+                    {
+                        Output.Information(false, ConsoleColor.Green, string.Format(CultureInfo.CurrentCulture, CommandLineResources.TestRunSummaryPassed, summary.TotalTests, summary.PassedTests, summary.FailedTests, summary.SkippedTests, GetFormattedDurationString(summary.TimeSpan), sd.Key.Split('\\').Last()));
+                    }
+                }
+                return;
+            }
+
+            foreach (var sd in this.sourceSummaryDictionary.ToArray())
```

Comment:
> foreach ( [](start = 11, length = 10)

Use one loop over the dictionary to calculate total information

Reply:
changed